glue/qt/widgets/histogram_widget.py

```diff
@@ -193,6 +182,20 @@ def component(self, component):
 
     @defer_draw
     def _set_attribute_from_combo(self, *args):
+        for d in self._data:
+            try:
+                component = d.get_component(self.component)
+            except:
+                continue
+            else:
+                break
```

Comment:
@ChrisBeaumont - the above is ugly - is there a preferred way to retrieve a component from a component ID if the dataset is not known? Should this be a method on ``DataCollection``? (or can one component ID in principle refer to components in separate dataset?)